classes/local/apibridge.php

```diff
@@ -1791,4 +1849,226 @@ public function get_duplicated_episodeid($workflowid) {
+
+        // Put events data in one place to make it simpler to use later.
+        $eventsaclchangeobject = new \stdClass();
+        // If there are vidoes.
+        if ($coursevideos &&  $coursevideos->error == 0) {
+            // Defining count will help with further process of result.
+            $eventsaclchangeobject->total = count($coursevideos->videos);
+            // Looping through videos.
+            foreach ($coursevideos->videos as $video) {
+                // Change the ACL of the event, by using making it visible for the new course.
+                $eventaclchange = $this->imported_events_acl_change($video->identifier, $courseid);
+                
+                // When there is an error, we sort them out in failed category.
+                if (!$eventaclchange) {
+                    // Adding video id to the failed element of the object.
+                    $eventsaclchangeobject->failed[] = $video->identifier;
+                    $result->error = 1;
+                    continue;
+                }
+
+                // We keep record of success items too. Makes it easy for later.
+                $eventsaclchangeobject->success[] = $video->identifier;
+            }
+        }
+        // Assigning that object to the result in one go.
+        $result->eventsaclchange = $eventsaclchangeobject;
+        
+        // Step 2: Update series ACL roles.
+        $result->seriesaclchange = $this->imported_series_acl_change($courseid, $seriesid, $userid);
+        
+        // When the Series ACL could not be changed.
+        if (!$result->seriesaclchange) {
+            $result->error = 1;
+        }
+
+
+        // Step 3: Assign seriesid to the new course in seriesmapping.
+        $result->seriesmapped = $this->map_imported_series_to_course($courseid, $seriesid);
+        if (!$result->seriesmapped) {
+            $result->error = 1;
+        }
+
+        // Finally return the result object, containing all the info about the whole process.
+        return $result;
+    }
+
+    /**
+     * Change and update the ACL of the series.
+     * @param int $courseid Course ID.
+     * @param string $seriesid Series ID.
+     * 
+     * @return bool
+     */
+    private function imported_series_acl_change($courseid, $seriesid, $userid) {
+        
+        // Reading acl from opencast server.
+        $api = new api();
+        $resource = '/api/series/' . $seriesid . '/acl';
+        $jsonacl = $api->oc_get($resource);
+
+        $acl = json_decode($jsonacl);
+
+        // When the acl could not be retreived.
+        if (!is_array($acl)) {
+            return false;
+        }
+
+        // Get the current defined roles.
+        $roles = $this->getroles();
+        foreach ($roles as $role) {
+            // Initialize the role object to have a better grip.
+            $roleobject = new \stdClass();
+
+            if (strpos($role->rolename, '[USERNAME]') !== false ||
```

Comment:
I don't think you need this if statement. Using the code within the if block should give you the same result for both branches.